packages/pilot/src/containers/RecipientDetails/Balance/index.js

```diff
@@ -0,0 +1,550 @@
+  }
+
+  handleFilterClick () {
+    const { onFilterClick } = this.props
+    const { dates } = this.state
+    if (isEmptyDates(dates)) {
+      onFilterClick(anyDateRange)
+    } else {
+      onFilterClick(dates)
+    }
+  }
+
+  handleOperationsPageChange (pageIndex) {
+    const { onPageChange } = this.props
+    onPageChange(pageIndex)
+  }
+
+  handleRequestCancelClick (requestIndex) {
+    const {
+      onCancelRequestClick,
+      requests,
+    } = this.props
+
+    onCancelRequestClick(requests[requestIndex].id)
+  }
+
+  handlePresetChange () {
+    this.setState({
+      showDateInputCalendar: true,
+    })
+  }
+
+  renderAnticipation () {
+    const {
+      anticipation: {
+        automaticTransfer,
+        available,
+        error,
+        loading,
+      },
+      t,
+    } = this.props
+
+    if (loading) {
+      return (
+        <span>
+          {t('pages.balance.anticipation_loading')}
+        </span>
+      )
+    }
+
+    if (error) {
+      return (
+        <span>
+          {t('pages.balance.anticipation_error')}
+          <Spacing size="tiny" />
+          <Tooltip
+            placement="rightMiddle"
+            content={t('pages.balance.anticipation_error_info')}
+          >
+            <IconInfo height={16} width={16} />
+          </Tooltip>
+        </span>
+      )
+    }
+
+    if (available === 0) {
+      return (
+        <span>
+          {t('pages.balance.no_anticipation')}
+        </span>
+      )
+    }
+
+    if (automaticTransfer) {
+      return (
+        <div>
+          <p>{t('pages.balance.anticipation_call')}</p>
+        </div>
+      )
+    }
+
+    return (
+      <span>
+        {t('pages.balance.available_anticipation')}
+        <strong> {formatAmount(available)} </strong>
+      </span>
+    )
+  }
+
+  render () {
+    const {
+      anticipation: {
+        available: availableAnticipation,
+        error: anticipationError,
+        loading: anticipationLoading,
+      },
+      balance: {
+        amount,
+        available: {
+          withdrawal,
+        },
+        outcoming,
+      },
+      currentPage,
+      dates,
+      disabled,
+      exporting,
+      itemsPerPage,
+      loading,
+      onAnticipationClick,
+      onCancelRequestClick,
+      onExport,
+      onPageCountChange,
+      onWithdrawClick,
+      pageSizeOptions,
+      search: {
+        operations,
+      },
+      t,
+    } = this.props
+
+    const { showDateInputCalendar } = this.state
+
+    const translateColumns = getColumnsTranslator(t)
+    const typesLabels = map(t, operationsTypesLabels)
+
+    const anticipationAction = {
+      disabled,
+      onClick: onAnticipationClick,
+      title: t('pages.balance.anticipation'),
+    }
+
+    const withdrawalAction = {
+      disabled,
+      onClick: onWithdrawClick,
+      title: t('pages.balance.withdraw'),
+    }
+
+    const filterDatesEqualCurrent = datesEqual(this.state.dates, dates) // eslint-disable-line
```

Comment:
Poderia remover este `eslint-disable-line`, só é preciso renomear a variável `this.state.dates`
na linha 300 
```
const {
      dates: stateDates,
      showDateInputCalendar,
    } = this.state
```
```suggestion
    const filterDatesEqualCurrent = datesEqual(stateDates, dates)
```